### PR TITLE
Activity log: Use API provided status

### PR DESCRIPTION
**Pull request by sirreal**
Previously, the status of an Activity item used for styling icons was calculated in the components. The API now provides the status of Activity items. This PR removes the calculation logic and consumes the status from the API.

## Testing
1. Do tests pass?
1. Visit https://calypso.live/stats/activity?branch=update/activity-log/use-api-status
1. Do the Activity icons continue to have appropriate colors? This is being provided by the API.

### CODE REVIEW

File: client/my-sites/stats/activity-log-item/activity-icon.jsx [hunk 1]
```diff
@@ -1,55 +1,25 @@
...
 import classNames from 'classnames';
 import Gridicon from 'gridicons';
-import { head, split } from 'lodash';
 
 export default class ActivityIcon extends PureComponent {
-
 	static propTypes = {
-		activityName: PropTypes.string.isRequired,
 		activityIcon: PropTypes.string.isRequired,
+		activityName: PropTypes.string.isRequired,
```
Comment (ockham) [inline client/my-sites/stats/activity-log-item/activity-icon.jsx]: This prop doesn't seem to be used anymore.

File: client/my-sites/stats/activity-log-item/index.jsx [hunk 2]
```diff
@@ -196,7 +197,7 @@ class ActivityLogItem extends Component {
 			<div className={ classes } >
 				<div className="activity-log-item__type">
 					{ this.renderTime() }
-					<ActivityIcon { ...pick( log, [ 'activityName', 'activityIcon' ] ) } />
+					<ActivityIcon { ...pick( log, [ 'activityIcon', 'activityName', 'activityStatus' ] ) } />
```
Comment (ockham) [inline client/my-sites/stats/activity-log-item/index.jsx]: Is `ActivityIcon` still using `activityName`? (Cf https://github.com/Automattic/wp-calypso/pull/18179/files#r140901125)

Comment (ockham) [reply client/my-sites/stats/activity-log-item/index.jsx]: If it doesn't, the destructured `pick` seems a bit overkill for two props. Maybe destructure them from `log` further up and pass `activityIcon` and `activityStatus` explicitly here?

File: client/my-sites/stats/activity-log-item/activity-icon.jsx (hunk 1 above)
Comment (sirreal) [reply client/my-sites/stats/activity-log-item/activity-icon.jsx]: Thanks for spotting that!

File: client/my-sites/stats/activity-log-item/index.jsx (hunk 2 above)
Comment (sirreal) [reply client/my-sites/stats/activity-log-item/index.jsx]: Definitely!

**Pull request merged**